### DISCUSSION TITLE
Is the custom endpoint cached?

### DISCUSSION THREAD

**Discussion by ghost** [directus/directus#7397] [Archive]
I'm trying to develop a custom endpoint to learn how to do. I have noticed that response time of custom endpoint do not vary editing CACHE_ENABLED on .env configuration. Standard endpoint instead have response time very different beetween cache enabled or disabled.

Custom endpoint aren't served by cache, or I must setup my custom endpoint properly to enable cache? If so, how?

Thank you!

**Comment by tylerforesthauser** [comment]
The magic is right here: https://github.com/workatease/directus-api/blob/master/extensions/endpoints/register/index.js#L8

Many thanks to @workatease for this current workaround 👍

**Reply by ghost** [reply]
I'm sorry, but reading code I don't understand how to integrate cache in my custom endpoint.

Only for testing purpose I have tried following code inside my custom endpoint:

```
module.exports = function registerEndpoint(router, { exceptions, database }) {
	
    const { ServiceUnavailableException } = exceptions;
	const asyncHandler = require("directus/dist/utils/async-handler").default;
	const response = require("directus/dist/middleware/respond").respond;

	router.get(
		"/hello",
		asyncHandler(async (req, res, next) => {
			setTimeout(function() {
				res.locals.payload = { result: 'hello' };
				return next();
			}, 1000);
		}),
		response
	  );


	router.get('/world', (req, res, next) => {
		setTimeout(function() {
			res.json({result:'hello'});
		}, 1000);
	});

};
```

I would like that, in case I call /hello endpoint, if result is in cache, res.locals.payload will be served without execute again function. Instead whether I call /hello or i call /world, results will be served after timeout time that I set to simulate application logic.

Where I'm wrong?

**Reply by rijkvanzanten** [reply]
@tuxinside Currently, you only have the `response` middleware on `/hello`, so you won't see any caching on `/world`.

The first time you run `/hello`, the API will cache the res.locals.payload value based on the current user, headers, env var settings (https://github.com/directus/directus/blob/main/api/src/middleware/respond.ts). The second hit of the same endpoint, the API will check if a cache record exists for the specific request, and will respond with it immediately (it won't even hit your custom endpoint at all)
